src/services/staking_data_service.ts

```diff
@@ -41,15 +43,35 @@ export class StakingDataService {
         return epoch;
     }
 
+    public async getCurrentEpochWithFeesAsync(): Promise<EpochWithFees> {
+        const rawEpochWithFees: RawEpochWithFees | undefined = _.head(
+            await this._connection.query(queries.currentEpochWithFeesQuery),
+        );
+        if (!rawEpochWithFees) {
+            throw new Error('Could not find the current epoch.');
+        }
+        const epoch = stakingUtils.getEpochWithFeesFromRaw(rawEpochWithFees);
+        return epoch;
+    }
+
     public async getNextEpochAsync(): Promise<Epoch> {
         const rawEpoch: RawEpoch | undefined = _.head(await this._connection.query(queries.nextEpochQuery));
```

Comment:
Are we now over-fetching information for this query?

Reply:
nah, it runs a separate query that doesn't SUM over fills